cmd/vfsgendev/vfsgendev.go

```diff
@@ -0,0 +1,188 @@
+// vfsgendev is a convenience tool for using vfsgen in a common development configuration.
+package main
+
+import (
+	"bytes"
+	"flag"
+	"fmt"
+	"go/ast"
+	"go/build"
+	"go/parser"
+	"go/token"
+	"io"
+	"io/ioutil"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"text/template"
+)
+
+var (
+	sourceFlag = flag.String("source", "", "Specifies the http.FileSystem variable to use as source.")
+	nFlag      = flag.Bool("n", false, "Print the generated source but do not run it.")
+)
+
+const (
+	sourceTags = "dev"
+
+	outputFilename = "data_generate.go"
+	outputTags     = "!dev"
+)
+
+type source struct {
+	ImportPath   string
+	PackageName  string
+	VariableName string
+}
+
+var errInvalidFormat = fmt.Errorf("invalid format")
+
+// parseSourceFlag parses the "-source" flag. It must have "import/path".VariableName format.
+func parseSourceFlag(sourceFlag string) (source, error) {
+	// Parse sourceFlag as a Go expression, albeit a strange one:
+	//
+	// 	"import/path".VariableName
+	//
+	e, err := parser.ParseExpr(sourceFlag)
+	if err != nil {
+		return source{}, errInvalidFormat
+	}
+	se, ok := e.(*ast.SelectorExpr)
+	if !ok {
+		return source{}, errInvalidFormat
+	}
+	importPath, err := stringValue(se.X)
+	if err != nil {
+		return source{}, err
+	}
+	variableName := se.Sel.Name
+
+	// Import package to get its full import path and package name.
+	ctx := build.Default
+	ctx.BuildTags = strings.Fields(sourceTags)
+	bpkg, err := ctx.Import(importPath, ".", 0)
+	if err != nil {
+		return source{}, fmt.Errorf("can't import package %q: %v", importPath, err)
+	}
+
+	return source{
+		ImportPath:   bpkg.ImportPath,
+		PackageName:  bpkg.Name,
+		VariableName: variableName,
+	}, nil
+}
+
+// stringValue returns the string value of string literal e.
+func stringValue(e ast.Expr) (string, error) {
+	lit, ok := e.(*ast.BasicLit)
+	if !ok {
+		return "", errInvalidFormat
+	}
+	if lit.Kind != token.STRING {
+		return "", errInvalidFormat
+	}
+	return strconv.Unquote(lit.Value)
+}
+
+var t = template.Must(template.New("").Funcs(template.FuncMap{
+	"quote": func(s string) string {
+		return strconv.Quote(s)
+	},
+}).Parse(`package main
+
+import (
+	"log"
+
+	"github.com/shurcooL/vfsgen"
+
+	{{.ImportPath | quote}}
+)
+
+func main() {
+	err := vfsgen.Generate({{.PackageName}}.{{.VariableName}}, vfsgen.Options{
+		PackageName:  {{.PackageName | quote}},{{with .BuildTags}}
+		BuildTags:    {{. | quote}},{{end}}
+		VariableName: {{.VariableName | quote}},
+	})
+	if err != nil {
+		log.Fatalln(err)
+	}
+}
+`))
```

Comment:
> I agree that the template is not near the context.. but it makes it easier for me to read because it's more apparent it's a template and not Go code as part of the primary file.

To help me understand this better, what does a string literal look like for you with your code highlighting scheme?

For me, it looks very distinct. I wonder if that's why I find having the template inline more readable.

http://virtivia.com:27080/1bk95aj57xsi5.png